### PR TITLE
Create hasContract() function for protocol

### PR DESCRIPTION
**Pull request by patiee**

### CODE REVIEW

File: src/protocols/tendermint/contract.js [hunk 1]
```diff
@@ -1,12 +1 @@
...
-  }
-
-  validate() {
-    return {
-      valid: true,
-      error: null,
-    }
-  }
-}
+module.exports = class TendermintContract {}
```
Comment (pienkowb) [inline src/protocols/tendermint/contract.js:1]: Are we able to remove this file entirely and still make it work?

Comment (patiee) [reply src/protocols/tendermint/contract.js:1]: no, then we have validation error

File: src/subgraph.js [hunk 2]
```diff
@@ -158,11 +158,19 @@ At least one such handler must be defined.`,
...
+
     return validation.validateContractValues(manifest, protocol)
   }
 
   // Validate that data source names are unique, so they don't overwrite each other.
-  static validateUniqueDataSourceNames(manifest) {
+  static validateUniqueDataSourceNames(manifest, protocol) {
+    if (!protocol.hasContract()){
+      return immutable.List()
+    }
```
[the remark this answers is not in the document]
Comment (patiee) [reply src/subgraph.js:172]: hmm yes that's unnecessary

File: src/protocols/tendermint/contract.js (hunk 1 above)
Comment (patiee) [reply src/protocols/tendermint/contract.js]: You were right, it works without `contract.js` file

**Review by pienkowb — COMMENTED** [review]
We need to return `null` [here](https://github.com/figment-networks/graph-cli/blob/82ebbbbb155a28957bc0323ab1c766137ddd6ea0/src/protocols/index.js#L198) and we're good to go 👍

**Pull request merged**